syntax = "proto3";

enum Role {
    ADMIN  = 0;
    EDITOR = 1;
    USER   = 2;
}

message UserInfo {
    int32  user_id = 1;
    string name    = 2;
    Role   role    = 3;
}

message ChatPacket {
    int32    user_id   = 1;
    string   content   = 2;
    UserInfo user_info = 3;
}